src/Management/Model/DataSource.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Doofinder\Management\Model;
+
+use Doofinder\Shared\Interfaces\ModelInterface;
+
+/**
+ * Model with data of a given data source
+ */
+class DataSource implements ModelInterface
+{
+    /**
+     * @var array<DataSourceOption>
+     */
+    const FQCN_OPTIONS = [
+        'file' => DataSourceOptionFile::class,
+        'shopify' => DataSourceOptionShopify::class,
+        'bigcommerce' => DataSourceOptionBigCommerce::class,
+        'ekm' => DataSourceOptionEkm::class,
+        'magento2' => DataSourceOptionMagento::class
+    ];
+
+    /**
+     * @var DataSourceOption
+     */
+    private $options;
+
+    /**
+     * @var string
+     */
+    private $type;
+
+    /**
+     * @param DataSourceOption $options
+     * @param string $type
+     */
+    private function __construct($options, $type)
+    {
+        $this->options = $options;
+        $this->type = $type;
+    }
+
+    /**
+     * @param array $data
+     * @return DataSource
+     */
+    public static function createFromArray(array $data)
+    {
+        /** @var class-string<DataSourceOption> $fqcn */
+        $fqcn = self::FQCN_OPTIONS[$data['type']];
```

Comment:
Could be possible that `$data['type']` does not exists on `FQCN_OPTIONS` and this code fails?

Reply:
It's not possibl because there are 5 different types, included in FQCN_OPTIONS constant.

Reply:
Ok, thanks :-)